src/codegen.c

```diff
@@ -447,16 +453,30 @@ static void codegen_expr(CodegenContext *ctx, Node *expr) {
 
   /// Literal expression. Only integer literals are supported for now.
   case NODE_LITERAL:
-    if (expr->literal.type != TK_NUMBER) DIAG(DIAG_SORRY, expr->source_location, "Emitting non-integer literals not supported");
+    if (expr->literal.type == TK_NUMBER) expr->ir = ir_immediate(ctx, expr->type, expr->literal.integer);
+    else if (expr->literal.type == TK_STRING) {
+      char buf[48] = {0};
+      static size_t string_literal_count = 0;
+      int len = snprintf(buf, 48, "__str_lit%zu", string_literal_count++);
+      // TODO: Two options. Since we can't currently assign to static
+      // variables with compile-time known constants, it means that
+      // strings go uninitialised... So what would really be ideal
+      // is if we could just, ya know, do that. (i.e. `.string` as
+      // directive or asciz or whathaveyou). Our other option is to
+      // emit (imm+store) pairs for every byte...
+      expr->ir = ir_create_static(ctx, expr->type, as_span(string_create(buf)));
+    }
+    else DIAG(DIAG_SORRY, expr->source_location, "Emitting non-integer literals not supported");
```

Comment:
Since literals use a `TokenType` as their type field, I wasn't sure how to actually print it. Thinking on it now, I probably could just print the entire type with `%T`. Either way, a somewhat scuffed implementation of this is found in 9b719820e918f2c58df280958b51132a42e43810

Reply:
That is true, though I’d hope that sema would have set the type, so you should just be able to do `%T` using `expr->type`

Reply:
Implemented in 974bdff6f8a86e336a4c723f70c67a59867fa206